mantidproject/mantid
# Indirect Data Analysis IqtFitMultiple - Remove temporary workspace

IqtFitMultiple no longer produces an unwanted temporary workspace when it is run. _This was originally thought to be caused by ProcessIndirectFitParameters_ but is actually caused by IqtFitMultiple.

**To test:**
- Turn on show invisible workspaces (View > Preferences > Mantid > options > `Show Invisible Workspaces`)
- Run the following script:

```
#Load in iqt data
input_ws = Load(Filename='iris26176_graphite002_iqt.nxs')
function = r'name=LinearBackground,A0=0.027668,A1=0,ties=(A1=0);name=UserFunction,Formula=Intensity*exp(-(x/Tau)^Beta),Intensity=0.972332,Tau=0.0247558,Beta=1;ties=(f1.Intensity=1-f0.A0)'

#run IqtFitMultiple
result, params, fit_group = IqtFitMultiple(InputWorkspace=input_ws, Function=function, FitType='1S_s', StartX=0, EndX=0.2, SpecMin=0, SpecMax=16)
```
- Ensure no `__TMP` workspaces are created in the in the ADS
- 5 workspaces should be in the ADS:

![image](https://cloud.githubusercontent.com/assets/13132128/16041647/7c04e93e-322e-11e6-9637-7bc62bfc6574.png)
- input_ws is the loaded data
- The additional 4 are generated by the algorithm and are expected

Fixes #15342

[RELEASE NOTES](https://github.com/mantidproject/mantid/blob/7b5fd0c57af694ea40af4ef5abf5a1581b3d0236/docs/source/release/v3.8.0/indirect_inelastic.rst#bugfixes)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
